src/dswx_sar/region_growing.py

```diff
@@ -0,0 +1,329 @@
+import os
+import logging
+import time
+import mimetypes
+
+import numpy as np
+from scipy import ndimage
+from joblib import Parallel, delayed
+
+from dswx_sar import dswx_sar_util
+from dswx_sar.dswx_runconfig import RunConfig, _get_parser
+
+
+logger = logging.getLogger('dswx_s1')
+
+
+def region_growing(likelihood_image,
+                   initial_seed=0.6,
+                   relaxed_threshold=0.45,
+                   maxiter=200):
+    """The regions are then grown from the seed points to adjacent
+    points since it covers the relaxed_threshold values.
+
+    Parameters
+    ----------
+    likelihood_image : numpy.ndarray
+        fuzzy image with values [0, 1]representing
+        likelihood of water where 0 is 0% and 1 is 100%
+    initial_seed : float
+        Initial threshold [0 - 1] used to classify
+        `fuzz_image` into water and non-water pixels.
+        If a pixel values exceeds `initial_seed`
+        then it is classified as water, otherwise it is
+        classified as non-water.
+    relaxed_threshold : float
+        relaxed threshold to be used for transient area
+        between water and non-water. Any pixel value
+        greater than threshold classified as water,
+        otherwise it's classified as non-water.
```

Comment:
```suggestion
        `fuzz_image` into feature and non-feature pixels.
        If a pixel values exceeds `initial_seed`
        then it is classified as feature, otherwise it is
        classified as non-feature.
    relaxed_threshold : float
        relaxed threshold to be used for transient area
        between feature and non-feature. Any pixel value
        greater than threshold classified as feature,
        otherwise it's classified as non-feature.
```
More replacements of `water` with `feature`.